notebooks/deepsd_data_prep.ipynb

```diff
@@ -0,0 +1,1235 @@
+    "    \"TRAIN_PERIOD_START\": \"1981\",\n",
+    "    \"TRAIN_PERIOD_END\": \"2010\",\n",
+    "    \"VARIABLES\": [\"tasmax\", \"tasmin\", \"pr\"],\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c0ed651f-34e5-4e5d-90df-26cd628617c6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "with Flow(name=\"prepare_obs_input_and_label\") as flow:\n",
+    "    obs = Parameter(\"OBS\")\n",
+    "    train_period_start = Parameter(\"TRAIN_PERIOD_START\")\n",
+    "    train_period_end = Parameter(\"TRAIN_PERIOD_END\")\n",
+    "    variables = Parameter(\"VARIABLES\")\n",
+    "    output_degree = Parameter(\"OUTPUT_DEGREE\")\n",
+    "\n",
+    "    # go from 721 x 1440 to 720 x 1440\n",
+    "    shifted_grid = build_grid_spec_task(output_degree=0.25)\n",
+    "    shifted_grid_obs = shift_obs_grid_task(\n",
+    "        obs=obs,\n",
+    "        train_period_start=train_period_start,\n",
+    "        train_period_end=train_period_end,\n",
+    "        variables=variables,\n",
+    "        output_degree=0.25,\n",
+    "        gcm_grid_spec=shifted_grid,\n",
+    "        chunking_approach=\"full_space\",\n",
+    "    )\n",
+    "\n",
+    "    # coarsen to desired resolution\n",
+    "    gcm_grid_spec = build_grid_spec_task(output_degree=output_degree)\n",
+    "    coarse_obs = coarsen_obs_task(\n",
+    "        ds_obs_full_space=shifted_grid_obs,\n",
+    "        obs=obs,\n",
+    "        train_period_start=train_period_start,\n",
+    "        train_period_end=train_period_end,\n",
+    "        variables=variables,\n",
+    "        output_degree=output_degree,\n",
+    "        gcm_grid_spec=gcm_grid_spec,\n",
+    "        chunking_approach=\"full_space\",\n",
+    "    )\n",
+    "\n",
+    "    # interpolate it back to 2x higher res\n",
+    "    gcm_grid_spec = build_grid_spec_task(output_degree=output_degree / 2.0)\n",
+    "    coarse_obs = interpolate_obs_task(\n",
+    "        ds=coarse_obs,\n",
+    "        obs=obs,\n",
+    "        train_period_start=train_period_start,\n",
+    "        train_period_end=train_period_end,\n",
+    "        variables=variables,\n",
+    "        output_degree=output_degree / 2.0,\n",
+    "        gcm_grid_spec=gcm_grid_spec,\n",
+    "        chunking_approach=\"full_space\",\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e87edd88-f04e-4e62-b324-a9228cc3af18",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "for output_degree in [0.5, 1.0, 2.0]:\n",
+    "    run_hyperparameters[\"OUTPUT_DEGREE\"] = output_degree\n",
+    "    flow.run(parameters=run_hyperparameters)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "28209eb8-d7ce-45ba-833a-ef34c76c85b7",
+   "metadata": {},
+   "source": [
+    "## Also coarsen elevation\n",
+    "\n",
+    "Elevation data is used as auxilliary input features to the deep sd model. The\n",
```

Comment:
What other input features might be useful here to include? Like, if we were to revisit this step at a later date, what other features should be top of the list to explore?